### PR TITLE
Meeting context bot - csharp and nodejs

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**
**Please do not review client side as it has already been reviewed in previous PR** - https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/406

### CODE REVIEW

File: samples/meetings-context-app/csharp/MeetingContextApp/AppManifest/manifest.json [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
-    "short": "Meeting Tab Context",
-    "full": "This sample shows the contents of a tab context object in a meeting tab."
+    "short": "Meeting Context App",
+    "full": "Meeting Context App"
   },
   "description": {
-    "short": "Tab context for meeting tab.",
-    "full": "This sample shows the contents of a tab context object in a meeting tab."
+    "short": "Tab and bot context for meeting.",
+    "full": "This sample shows the contents of a meeting context object in a meeting tab and using bot API."
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/MeetingContextApp/AppManifest/manifest.json:23]: I think, we are showing data in bot as well ?

File: samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+                    await turnContext.SendActivityAsync(MessageFactory.Text(formattedString), cancellationToken);
+                }
+                else
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text(commandString), cancellationToken);
+                }
+
+            }
+
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs:52]: We should remove empty extra lines ?

File: samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var welcomeText = "Hello and Welcome!";
+
+            await turnContext.SendActivityAsync(MessageFactory.Text(welcomeText), cancellationToken);
+            await turnContext.SendActivityAsync(MessageFactory.Text(commandString), cancellationToken);
+        }
+
+        private string GetFormattedSerializeObject (object obj)
+        {
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs:64]: We can add method summary as well?

File: samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            var formattedString = "";
+            foreach (var meetingDetails in obj.GetType().GetProperties())
+            {
+                var detail = meetingDetails.GetValue(obj, null);
+                var block = $"<b>{meetingDetails.Name}:</b> <br>";
+                var temp = "";
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs:70]: what is temp ?
Can we use some meaningful name here ?

File: samples/meetings-context-app/csharp/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
+---
+page_type: sample
+description: This sample shows the contents of meeeting tab context object in a meeting tab and meeting context details where user can see the details of participants and meeting via bot.
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/README.md:3]: Description is kind of messy
This sample shows the contents of meeeting tab context object in a **meeting tab and meeting context details** where user can see the details of participants and meeting via bot.

Please refactor it, if we can.

Also, spelling mistake here "**meeeting** "

File: samples/meetings-context-app/csharp/README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  createdDate: "02/08/2022 12:26:57 PM"
+---
+
+# Meeting context
+
+This sample shows the contents of meeeting tab context object in a meeting tab and shows the output of Meeting's API `TeamsInfo.GetMeetingParticipantAsync` and `TeamsInfo.GetMeetingInfoAsync` using bot commands.
+
+![meeting tab context](MeetingContextApp/Images/meetingTabContext.png)
+![Meeting context](MeetingContextApp/Images/MeetingContext.png)
+![Participant context](MeetingContextApp/Images/ParticipantContext.png)
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/csharp/README.md:21]: Please verify if these images are visible properly in UI>

File: samples/meetings-context-app/nodejs/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+---
+page_type: sample
+description: This sample shows the contents of meeeting tab context object in a meeting tab and meeting context details where user can see the details of participants and meeting via bot.
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/nodejs/README.md:3]: spelling mistake "**meeeting** "

Also check the description 
meeeting **tab context object in a meeting tab and meeting context details**

File: samples/meetings-context-app/nodejs/README.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+createdDate: "01-07-2022 17:00:25"
+---
+
+# Meeting context
+
+This sample shows the contents of meeeting tab context object in a meeting tab and shows the output of Meeting's API `TeamsInfo.getMeetingParticipant` and `TeamsInfo.getMeetingInfo` using bot commands.
+
+![meeting tab context](Images/meetingTabContext.png)
+![Meeting context](Images/MeetingContext.png)
+![Participant context](Images/ParticipantContext.png)
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/nodejs/README.md:21]: Verify if these images are visible properly.

File: samples/meetings-context-app/nodejs/server/bots/teamsBot.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Gets the serialize formatted object string.
+     * @param {object} obj Incoming object needs to be formatted.
+     * @returns Formatted string.
+     */
+    async formatObject(obj) {
+        let formattedString = "";
+        Object.keys(obj).forEach((key) => {
+            var block = `<b>${key}:</b> <br>`;
+            var temp = "";
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/nodejs/server/bots/teamsBot.js:57]: rename temp to some meaningful name

File: samples/meetings-context-app/nodejs/server/package.json [hunk 10]
```diff
@@ -0,0 +1,40 @@
+{
+    "name": "teams-bot",
+    "version": "1.0.0",
```
Comment (ChetanSharma-msft) [inline samples/meetings-context-app/nodejs/server/package.json:3]: Are these values correct ?

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Reviewed and added my comments

File: samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs (hunk 3 above)
Comment (HunaidHanfee-MSFT) [reply samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs:64]: Added

File: samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs (hunk 4 above)
Comment (HunaidHanfee-MSFT) [reply samples/meetings-context-app/csharp/MeetingContextApp/Bots/MeetingContextBot.cs:70]: It is there to just hold string temporarily. Updated it as storeTemporaryFormattedString

File: samples/meetings-context-app/nodejs/README.md (hunk 7 above)
Comment (HunaidHanfee-MSFT) [reply samples/meetings-context-app/nodejs/README.md:3]: Updated.

File: samples/meetings-context-app/nodejs/server/bots/teamsBot.js (hunk 9 above)
Comment (HunaidHanfee-MSFT) [reply samples/meetings-context-app/nodejs/server/bots/teamsBot.js:57]: Done

File: samples/meetings-context-app/nodejs/server/package.json (hunk 10 above)
Comment (HunaidHanfee-MSFT) [reply samples/meetings-context-app/nodejs/server/package.json:3]: Yes

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**